src/jsonapi/jsonapi-generator.py

```diff
@@ -157,40 +157,92 @@ def processFile(file):
+							pName = defNameNode.text
+						elif declNameNode is not None and declNameNode.text:
+							pName = declNameNode.text
+				# --- End of Name Extraction ---
+
+				# Skip parameter if no name could be determined
+				if not pName:
+					print(f"Warning: Could not determine parameter name for param type [{pType}] in {refid}. Skipping.")
+					continue 
+
+				# Create and populate the MethodParam object
+				mp = MethodParam()
+				mp._name = pName # Assign the determined name FIRST
+				mp._type = pType # Assign the initial type
+				
+				tmpDefvalNode = tmpPE.find('defval')
+				mp._defval = getText(tmpDefvalNode) if tmpDefvalNode is not None else ''
+				mp._defval = ' '.join(mp._defval.split()) # Clean default value whitespace
+
+				# --- Type Cleaning and Callback Detection ---
+				cleaned_pType = pType 
+				if cleaned_pType.startswith('const '): 
+					cleaned_pType = cleaned_pType[6:]
+
+				if cleaned_pType.startswith('std::function'):
+					if cleaned_pType.endswith('&'): 
+						cleaned_pType = cleaned_pType[:-1] 
+					
 					if pName.startswith('multiCallback'):
 						mp._isMultiCallback = True
 						hasMultiCallback = True
+						callbackName = pName
+						callbackParams = cleaned_pType
 					elif pName.startswith('callback'):
 						mp._isSingleCallback = True
 						hasSingleCallback = True
-					callbackName = pName
-					callbackParams = pType
-				else:
-					pType = pType.replace('&', '').replace(' ', '')
-				
-				# Apparently some xml declarations include new lines ('\n') and/or multiple spaces
-				# Strip them using python magic
-				pType = ' '.join(pType.split())
-				mp._defval = ' '.join(mp._defval.split())
-
-				mp._type = pType
-				mp._name = pName
-
+						callbackName = pName 
+						callbackParams = cleaned_pType 
+					
+					mp._type = cleaned_pType # Assign cleaned signature for callbacks
+
+				else: # Not a callback, just clean the regular type
+					# Don't remove '&' for regular types during cleaning here,
+					# it might be needed later for serialization/declaration.
+					# Just clean whitespace.
+					# cleaned_pType = cleaned_pType.replace('&', '').replace(' ', '') 
+					cleaned_pType = ' '.join(cleaned_pType.split())
+					mp._type = cleaned_pType # Assign cleaned type
+				# --- End of Type Cleaning ---
+
+				# Add the parameter to the map and ordered list
 				paramsMap[pName] = mp
 				orderedParamNames.append(pName)
 
 			for tmpPN in memberdef.findall('.//parametername'):
-				tmpParam = paramsMap[tmpPN.text]
+				# Add Try/Except block to handle cases where parameter name from docs doesn't match params from signature
+				try:
+					tmpParam = paramsMap[tmpPN.text]
+				except KeyError:
+					print(f"Warning: Parameter '{tmpPN.text}' found in Doxygen description for {refid}, but not in function signature parameters {list(paramsMap.keys())}. Skipping direction assignment.")
+					continue # Skip to the next parametername tag
+
```

Comment:
AFAIR we should fail in this case, in addition to print the error, so compilation fail and developers notice ASAP they screw-up something, am I wrong?